gobblin-utility/src/main/java/org/apache/gobblin/util/JobLauncherUtils.java

```diff
@@ -197,6 +197,16 @@ public static void cleanTaskStagingData(State state, Logger logger) throws IOExc
           throw new IOException("Clean up output directory " + outputPath.toUri().getPath() + " failed");
         }
       }
+
+      if (state.contains(ConfigurationKeys.ROW_LEVEL_ERR_FILE) && state.getPropAsBoolean(ConfigurationKeys.USE_DATASET_LOCAL_WORK_DIR, false)) {
```

Comment:
We should avoid leaking details about dataset-specific local work dir all over the place. This change seems unnecessary too.